code/modules/mob/living/simple_animal/bot/bot.dm

```diff
@@ -371,18 +371,37 @@ The proc would return a human next to the bot to be set to the patient var.
 Pass the desired type path itself, declaring a temporary var beforehand is not required.
 */
 /mob/living/simple_animal/bot/proc/scan(scan_type, old_target, scan_range = DEFAULT_SCAN_RANGE)
-	var/final_result
+	var/turf/T = get_turf(src)
+	if(!T)
+		return
+
+	for(var/scan in T.GetAtmosAdjacentTurfs(1))//Let's see if there's something right next to us first!

```

Comment:
store```T.GetAtmosAdjacentTurfs(1)``` in a list var, and then remove that var from the ```shuffle(view(scan_range,src))```, no point checking 4(8?) tiles again